### DISCUSSION TITLE
Proteins not bonded together or living in two separate PDB files

### DISCUSSION THREAD

**Discussion by gcroci2** [DeepRank/deeprank2#603] [Q&A] [answered]
A potential user asked me how to handle data in case 1) the two proteins live in two different PDB files, or 2) they are in the same PDB file but not bonded together (so I guess they are relatively far from each other). 

Case 1) is not currently handled by deeprank2, the user needs to take care of placing the two proteins together in the same PDB file. I think is fine this way, and thus it wouldn't make sense at all to add this feature. I just wanted to double-check with the other developers if they agree with me on this. @DaniBodor @cbaakman 

Regarding point 2) instead, I think this is a case users may encounter relatively often. Am I right that is the usual procedure to use software like PANDORA that places together in space the two molecules (in our case peptide + MHC) AS IF they bind together? @DarioMarzella Is it the same for Rosetta, AlphaFold2, etc.?
If not, then I guess that the two molecules can be placed in the same PDB file but as completely separate from each other. In such cases, would it be okay to just define a cut-off big enough to include the two molecules in the protein-protein interface query? Would it make sense at all from the domain point of view (since you won't have an "actual" interface being the molecules separated in the 3D space)? @DaniBodor @cbaakman @LilySnow

**Comment by cbaakman** [comment]
1) I agree

2) I think it wouldn't make sense to place two molecules in one PDB file if there's no real biological contact. Deeprank is supposed to work from contacts. (residues that lie next to each other)

**Comment by DaniBodor** [comment, marked as answer]
1) I agree it's best to keep things as they are. If proteins live in separate pdb files, I would be concerned about whether their position values are calculated taking each other into account or not (as opposed to e.g. centering each one individually). I think there is much more opportunity for erroneous or faulty data if we allow separate pdb files.

2) Similar to my response to #604, it all depends on what question the user is asking Deeprank2 to answer. Currently, we are equipped to answer two types of questions: 1) do these chains (in their current positions) bond or not, or 2) will this mutation have a functional consequence. I don't see how a chain that is really far away is relevant to either of these questions. For 1, the answer will almost certainly be "no, because they are two far away" and for 2, the far away chain is unlikely to have any bearing on the question.
That being said, there is nothing stopping users from using such data. By adjusting the values as you say, the questions _can_ be asked. However, similar to what @cbaakman says, I struggle to think of a situation where these questions bear relevance.

**Reply by gcroci2** [reply]
Great, we all agree on 1. ; regarding 2., the potential user was referring to cases in which you don't have a software (like PANDORA) that allows you to put together in the space the two proteins AS IF they bind, and thus you can only place the two proteins in the same PDB file, taking care of not placing them too far apart from each other to save on the memory side. But indeed, I think that in such cases the entire question that the predictor is asked to answer to should be likely revised.

**Reply by DaniBodor** [reply]
Yes, relative position matters (a lot) for DeepRank2. So if the question is: "here are two proteins, will they bind", this is not a task that we are set up to answer and I don't think it's in the scope of the project to go there either.
